### DISCUSSION TITLE
is this behavior corrrect?? client session keeps authenticated even if user signout by signout methods.

### DISCUSSION THREAD

**Discussion by takakikasuga** [nextauthjs/next-auth#12355] [Help]
## Reproduction URL
https://github.com/nextauthjs/next-auth-example

## behavior 
client session keeps authenticated even if user signout by signout methods.
https://next-auth-example.vercel.app/client-example

https://github.com/user-attachments/assets/02e5d807-2868-4cba-b9b0-1eb4cbae4a22

**Comment by nic-vo** [comment]
@takakikasuga you're right. This example is actually a bad example of client-side next-auth.

If you look at the [source](https://github.com/nextauthjs/next-auth-example/blob/main/app/client-example/page.tsx) for this page and the associated [app router layout](https://github.com/nextauthjs/next-auth-example/blob/main/app/layout.tsx) [component](https://github.com/nextauthjs/next-auth-example/blob/main/components/user-button.tsx#L13), you'll see that the `<header>` sign out button doesn't trigger any updates to the `page.tsx` component upon redirect from the OAuth provider sign in flow.

You're technically still signed out, though. To verify this:
1) Sign in, complete flow, then click sign out button. `<header>` should refresh to "Sign in" button again.
2) Open a new tab and click to it
3) Return to next auth client example page. The `<SessionProvider />` has a configurable "check session on window refocus" function that will trigger and update the component as expected.

The problem with this example is that the none of the sign in or sign out functionality takes place in the `page.tsx` client component. It takes place in the initial server component render that then gets cached client-side. The `<header>` sign out button has no way of forcing re-render of the `page.tsx` because I think it's scoped to the `layout.tsx`. Technically, it still works as expected though.
